wf-galleon/WFLY-11989_Galleon_Maven_Repository.adoc

```diff
@@ -0,0 +1,90 @@
+= WFLY-11989: wildfly-dist should generate and deploy a zipped maven repo for galleon
+:author:            Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+Offer the ability to package all Galleon required artifacts to provision WildFly server.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11989[WFLY-11989]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-891[EAP7-891]
+
+* https://issues.jboss.org/browse/EAP7-1218[EAP7-1218]
+
+* https://issues.jboss.org/browse/WFGP-132[WFGP-132] (Obsolete, not followed approach)
+
+* https://issues.jboss.org/browse/WFGP-135[WFGP-135]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* OpenShift
+
+=== Other Interested Projects
+
+* WildFly Galleon
+
+* Offliner: https://github.com/release-engineering/offliner
+
+== Requirements
+
+In order to address this requirement, we are defining a new WildFly Maven plugin (_org.wildfly.galleon-plugins:wildfly-galleon-maven-plugin_)
+goal named _generate-all-artifacts-list_.
+This goal aggregates all offliner files that have been generated when each Wildfly galleon feature-pack is built.
```

Comment:
Are these already being generated? I quick check of galleon-pack/target in my current wildfly checkout doesn't show one.

Reply:
@bstansberry , this is coming. I just opened https://github.com/wildfly/wildfly-core/pull/3751
to make core to generate the offliner that will be aggregated in wildfly-dist.

Reply:
during each galleon pack build (or precisely in another maven lifecycle phase after galleon pack build) will be generated artifact list (=offline file) for given pack. Those will be aggregates as described above.
On the product level we will assert that EAP maven repository contains all artifacts mentioned in the aggregated artifact list (it would be subset of maven repository, which contains additional artifacts required eg. for develepment)

Reply:
@jfdenise  BTW - those additional artifacts which are not part of any galleon pack you may need as well - eg. jboss-ejb-client-legacy - not sure whether it is part of GP or not